### ISSUE TITLE
When used in Next.js, I get a "Cannot use import statement outside a module" error.

### ISSUE THREAD

**Issue by cateiru** [sators/react-password-checklist#18]
**Describe the bug**

When used with Next.js, a "Syntax Error: Cannot use import statement outside a module" error occurs during server-side rendering.

repository: https://github.com/cateiru/react-password-checklist-nextjs

**To Reproduce**
1. clone repo: https://github.com/cateiru/react-password-checklist-nextjs
2. `yarn install`
3. `yarn dev`
4. access localhost:3000

Also, the build fails

```log
❯❯❯ yarn build
yarn run v1.22.11
$ next build
info  - Checking validity of types
info  - Creating an optimized production build
info  - Compiled successfully

> Build error occurred
/react-password-checklist-nextjs/node_modules/react-password-checklist/dist/index.js:27
import React, { useState, useEffect } from "react";
^^^^^^

SyntaxError: Cannot use import statement outside a module
    at Object.compileFunction (node:vm:352:18)
    at wrapSafe (node:internal/modules/cjs/loader:1031:15)
    at Module._compile (node:internal/modules/cjs/loader:1065:27)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:94:18)
    at Object.782 (/react-password-checklist-nextjs/.next/server/pages/index.js:25:59)
    at __webpack_require__ (/react-password-checklist-nextjs/.next/server/webpack-runtime.js:25:42) {
  type: 'SyntaxError'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Expected behavior**

**Screenshots**

![スクリーンショット 2022-01-25 120222](https://user-images.githubusercontent.com/24271196/150904119-1734d2e7-6cbf-48a5-ba0a-23335689f764.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu
 - Browser: Chrome
 - Version: 97.0.4692.99

**Smartphone (please complete the following information):**

Haven't tried

**Additional context**

**Comment by cateiru** [comment]
Importing only on the client side using next/dynamic fixed the error.

```ts
import dynamic from 'next/dynamic';
const PasswordChecklist = dynamic(() => import('react-password-checklist'), {
  ssr: false,
});
```

**Comment by sators** [comment]
Yes, this was mentioned on https://github.com/sators/react-password-checklist/issues/15#issuecomment-999797864.

In digging around, I'm curious if you would try the following and report back @cateiru - after installing via yarn, edit the `node_modules/react-password-checklist/package.json` and add `"type": "module",` and see if that resolves the issue.  If so, I'll build that into the next release.

**Comment by cateiru** [comment]
@sators 
Sorry, I overlooked the fact that the issue stood in the past.
This issue will be closed as it was resolved in next/dynamic. Thank you very much.

**Comment by zanedev** [comment]
@sators I confirmed adding type="module" fixed it for me in a remix project. Can you add it to the next release? Thanks

**Comment by sators** [comment]
This is now available in v1.8
